### ISSUE TITLE
Unable to use Vite's Hot Reload when running Sail inside Docker on WSL2

### ISSUE THREAD

**Issue by woganmay** [laravel/framework#45938]
- Laravel Version: 9.48.0
- PHP Version: 8.1.13
- Database Driver & Version: Irrelevant

### Description:
I'm using WSL2 and Laravel Sail inside Docker. When Vite starts its Hot server, it binds to 'localhost' by default, which makes it inaccessible from outside the container. If I change server.host to 0.0.0.0 it becomes accessible from my Windows browser, but Chrome (and Firefox) treat 0.0.0.0 as an invalid hostname.

### Steps To Reproduce:

1. Set up Windows and WSL2, install Docker
2. Create a new Laravel 9 project using Vite
3. Start the Vite server with --host=0.0.0.0 inside the container
4. Observe that the `@vite` helper renders out http://0.0.0.0:5173/@vite/client in the browser
5. Observe that Chrome (109.0.5414.120) blocks the GET request to that URL.

### Fix:

I got around this by creating a ViteOverride class, binding it to `Illuminate\Foundation\Vite` and overriding the hotAsset method to this:

```
$baseUrl = file_get_contents($this->hotFile());
if (getenv('LARAVEL_SAIL') == 1) $baseUrl = str_replace("0.0.0.0", "localhost", $baseUrl);
return rtrim($baseUrl) . '/' . $asset;
```

Not sure if that's the best approach though, or if others are having this issue.

**Comment by driesvints** [comment]
I guess this is a duplicate of https://github.com/laravel/vite-plugin/issues/197 ?

**Comment by driesvints** [comment]
Closing this one.

**Comment by rklomp** [comment]
Having the same issue trying to run with sail in docker.
Can you elaborate on your fix where I should implement this?
Or is there a better solution available?

**Comment by sbaum2s** [comment]
> Solved for me by: [vitejs/vite#9155 (comment)](https://github.com/vitejs/vite/discussions/9155#discussioncomment-3344564)

This helped me solve the hot reload issue. But hot reload tihs way is so incredibly slow at least on my machine. I wait like 20-30 seconds for a simple change of a string. Running vite and the webserver outside WSL2 on a Windows machine gives me almost instant response, like 2-3 seconds. But then sail / docker has not point, since i would get a faster experience by re-running "npm run dev" or "npm run build" every time in changed something. I know its not efficient but much faster than hot reload via WSL2.
